### PR TITLE
add docs for shortnames

### PR DESCRIPTION
**Pull request by mckayward**

### CODE REVIEW

File: docs/guides/shortnames.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+
+Here are some examples that demonstrate how to use shortnames. For these
+examples, we'll assume your username is `fooster`, the project you've
+initialized in the current directory is called `my_project`, and the last job
+you ran in `my_project` is number `3`. Each example will show two code
+snippets: the first showing a command using a shortened name, and a second
+showing the full-length equivalent of the first command.
+
+
+## Job Name Examples
```
Comment (narenst) [inline docs/guides/shortnames.md:44]: This is great. Can you also add the list of subcommands where this shortened name would work? preferably with links to their usage pages?

**Review by narenst — COMMENTED** [review]
Can you also update the command usage pages? To make sure they show these new options?

**Comment by narenst** [comment]
I'm merging this now. Please create a new PR for other changes.

**Comment by mckayward** [comment]
Sounds good. I was just coming on here to say we should merge now and update later. I'll open another PR for the finishing touches

**Pull request merged**